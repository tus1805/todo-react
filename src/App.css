body {
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(to right, lightblue , pink);
  height: 100vh;
  margin: 0;
}

.App {
  width: 100vw;
  height: 100vh;
}

.form-container {
  padding: 20px 40px;
  border: 3px solid gray;
  border-radius: 10px;
  color: white;
  box-shadow: rgba(0, 0, 0, 0.56) 0px 22px 70px 4px;
  background-color: #607d8b;
  width: fit-content;
  margin: 0 auto;
  position: relative;
  top: 5%;
}

.form-group {
  color: white
}

.form-container form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.form-container a {
  margin: 10px;
  color: #00f3ff;
  font-weight: bold;
  text-decoration: none;
}
.form-container a:hover {
  text-decoration: underline;
  transition: 0.3s;
  cursor: pointer;
}

.form-container h1 {
  text-align: center;
  text-shadow: 2px 2px 4px midnightblue;
}

.form-container button {
  margin-top: 25px;
  margin-bottom: 5px;
  padding: 10px 15px;

  background-color: green;
  border: 1px solid darkgreen;
  color: white;
  border-radius: 6px;

  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;
}

.form-group {
  margin-bottom: 5px;
}

.form-group label {
  display: block;
  margin-bottom: 4px;
  color: white;
}

.form-group input {
  width: 200px;
  padding: 4px 10px;
}

.radio-group {
  align-self: flex-start;
}

.radio-group label:hover {
  cursor: pointer;
}

.checkbox-group {
  display: flex;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.error-message {
  display: block;
  color: yellow;
  width: 224px;
}

.error-message.success {
  display: block;
  color: green;
}

input#name {
  text-transform: capitalize;
}

input#username {
  text-transform: lowercase;
}

.header {
  display: flex;
  justify-content: space-between;
  margin: 10px;
  align-items: center
}
.header a {
  text-decoration: none;
  color: white;
  padding: 10px;
  font-size: 1.5rem;
  display: inline-block;
}
.header a:focus {
  text-decoration: none;
  color: white;
}
.header a:hover {
  text-decoration: underline;
}

.logout-button {
  background-color: white;
  border: 2px solid black;
  border-radius: 10px;
  width: 80px;
  height: 30px;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  outline: 2px gray solid;
  margin: 5px;
}
.logout-button:hover {
  background-color: lightseagreen;
  color: white;
}

.l-container {
  width: 100%;
  height: 100%;
}
.welcome-message {
  color: white;
  text-shadow: 1px 1px 10px deepskyblue;
}
.todo-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
.todo-list {
  margin: 10px;
  background: white;
  border-radius: 10px;
  border: 3px solid darkcyan;
  padding: 0 20px;
  min-width: 250px;
}
.todo-item {
  display: flex;
  width: 500px;
  margin: 10px;
  border-bottom: 1px solid black;
  padding-bottom: 8px;
}

.todo-item-taskname {
  flex-grow: 10;
  padding: 5px;
}
.todo-item:nth-child(4n + 1) {
  color: orchid
}
.todo-item:nth-child(4n + 2) {
  color: green
}
.todo-item:nth-child(4n + 3) {
  color: chocolate
}
.todo-item:nth-child(4n) {
  color: darksalmon
}
.button-update-task {
  display: none;
}

.button-update-user{
  display: none;
}

input#checkbox[type=checkbox]:checked + label{
  text-decoration: line-through;
}

.line-through {
  text-decoration: line-through;
}
.filter-option {
  margin: 10px;
}
.add-task-field {
  outline: none;
  border: 2px solid midnightblue;
  width: 300px;
  padding: 5px 5px;
  border-radius: 5px;
}
.button-add-task {
  background: rebeccapurple;
  color: white;
  border: 1px solid black;
  padding: 5px 10px;
  border-radius: 5px;
  font-weight: bold;
}
.button-clear-task {
  background: seagreen;
  color: white;
  border: 1px solid black;
  padding: 5px 10px;
  border-radius: 5px;
  font-weight: bold;
}
.button-update-task {
  background: blue;
  color: white;
  border: 1px solid black;
  padding: 5px 10px;
  border-radius: 5px;
  font-weight: bold;
}
.button-edit-task {
  background: yellow;
  color: darkcyan;
  border: 1px solid black;
  padding: 5px 10px;
  border-radius: 5px;
  font-weight: bold;
  margin-right: 5px;
}
.button-delete-task {
  background: red;
  color: white;
  border: 1px solid black;
  padding: 5px 10px;
  border-radius: 5px;
  font-weight: bold;
}
.filter {
  padding: 5px;
  font-size: 0.9rem;
  border-radius: 5px;
  background: deeppink;
  color: white;
  font-weight: 700;
}
.filter option {
  background: white;
  color: darkgray;
}
.project-control-container {
  display: flex;
}

table {
  border: 1px solid black;
}

table {
  width: 100%;
  padding: 10px;
  margin: 10px;
}
table td:nth-child(2) {
  text-align: center;
}
table th:nth-child(1) {
  text-align: left;
}

.user-info-table {
  margin: 20px;
  width: 300px;
}
.control-option {
  display: flex;
  justify-content: center;
}
.user-admin-control {
  display: flex;
  justify-content: center;
  align-items: center;
  height: calc(100vh - 60px);
}

.button-add-user.hide, .button-update-user.hide {
  display: none;
}

.button-add-user.show, .button-update-user.show {
  display: inline;
}

.button-add-user, .button-update-user {
  display: none;
}