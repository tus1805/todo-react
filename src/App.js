import "./App.css";
import SignIn from "./pages/SignIn";
import SignUp from "./pages/SignUp";
import ToDoList from "./pages/ToDoList"
import { adminAccount } from "./constants/error-message-signIn";
import { useEffect } from "react";
import { setItemWithLocal } from "./utils/process-data";
import ProjectAdmin from "./pages/Project-admin";
import UserAdmin from "./pages/User-admin";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Layout from "./components/Layout";

function App() {
  function setAdminAccount() {
    if (JSON.parse(localStorage.getItem("userdata"))) {
      return;
    }
    setItemWithLocal("userdata", [adminAccount]);
    return;
  }
  useEffect(() => {
    setAdminAccount();
  }, []);

  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Layout />}>
            <Route index element={<ToDoList />} />
            <Route path="project-admin" element={<ProjectAdmin />} />
            <Route path="user-admin" element={<UserAdmin />} />
          </Route>
          <Route path="sign-in" element={<SignIn />} />
          <Route path="sign-up" element={<SignUp />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
