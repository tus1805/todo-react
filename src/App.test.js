import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import renderer from "react-test-renderer";

describe("App component", () => {
  //Test 1: Ví dụ test quản lý state cơ bản
  it("starts with a count of 0", () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find("div.counter-value").text();
    expect(text).toEqual("Count: 0");
  });
  //Test 2: Ví dụ test tương tác của người dùng
  it("increments count by 1 when the increment button is clicked", () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find("button.increment");
    incrementBtn.simulate("click");
    const text = wrapper.find("div.counter-value").text();
    expect(text).toEqual("Count: 1");
  });
});

it("matches the snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});