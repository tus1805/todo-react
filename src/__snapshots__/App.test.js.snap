// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header"
    >
      <nav>
        <a
          href="/"
          onClick={[Function]}
        >
          Main
        </a>
        <a
          href="/project-admin"
          onClick={[Function]}
        >
          Project
        </a>
        <a
          href="/user-admin"
          onClick={[Function]}
        >
          User
        </a>
      </nav>
      <button
        className="logout-button"
        onClick={[Function]}
        type="button"
      >
        Log out
      </button>
    </div>
    <div
      className="l-container"
    >
      <div
        className="todo-container"
      >
        <h1
          className="welcome-message"
          id="welcome-message"
        >
          Hello
        </h1>
        <div
          className="todo-form"
        >
          <input
            className="add-task-field"
            id="add-task-field"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="button-add-task"
            onClick={[Function]}
            type="button"
          >
            Add
          </button>
          <button
            className="button-update-task"
            onClick={[Function]}
            type="button"
          >
            Update
          </button>
          <button
            className="button-clear-task"
            onClick={[Function]}
            type="button"
          >
            Clear
          </button>
        </div>
        <div
          className="filter-option"
        >
          <select
            className="filter"
            id="filter"
            name="filter"
            onChange={[Function]}
          >
            <option
              value="all"
            >
              All
            </option>
            <option
              value="done"
            >
              Done
            </option>
            <option
              value="undone"
            >
              Undone
            </option>
          </select>
        </div>
        <div
          className="todo-list"
          id="todo-list"
        />
      </div>
    </div>
  </div>
</div>
`;
